activerecord/test/cases/calculations_test.rb

```diff
@@ -1225,6 +1225,20 @@ def test_pluck_functions_without_alias
     )
   end
 
+  def test_column_types_typecast_with_duplicate_column_names
+    NumericData.create!(temperature: 1.0, decimal_number_big_precision: 5.23)
+    NumericData.create!(temperature: 2.0, decimal_number_big_precision: 4.56)
+    result = NumericData.pluck("sum(temperature)", "sum(decimal_number_big_precision)")
+
+    if current_adapter?(:PostgreSQLAdapter)
+      assert_equal Float, result.first.first.class
+      assert_equal BigDecimal, result.first.second.class
+    else
+      assert_equal Float, result.first.first.class
+      assert_equal Integer, result.first.second.class
```

Comment:
I'm a bit puzzled and would have expected `sum(decimal_number_big_precision)` to be mapped to `Float`. This is unrelated to the the changes of this PR also mapped to `Integer` without the changes provided here.